### PR TITLE
Mrollins/pr001

### PR DESCRIPTION
**Pull request by lazarus1331**
Infrastructure code for local Vagrant development box, integration server, Docker, and Docker Compose

### CODE REVIEW

**Comment by mscarey** [comment]
I saw the request for review. The recent changes look reasonable, as far as I understand them, but I still don't really understand how to contribute new data that can be linked to the TPJ data and put into production.

Even if the pull request doesn't solve that particular problem, I don't think the new commits will interfere with anything I'm doing.

**Comment by lazarus1331** [comment]
@mscarey Thank you for taking a look!  I guess I'm not sure which branch is our mainline, but I was attempting to merge back to the branch I originally checked out from.  I noticed that there were several commits from you, so I was hoping you could comment on whether those are safe to merge.  I realize the PR is rather large, and was also hoping for a comment like the one @mscarey provided about whether this would cause conflicts with their current work, so thank you for telling us that!
TLDR:
all the Dockers (Dockerfile, multiple docker-compose)
ansible code for managing postgresql and dependencies
vagrant environment for local development
integration environment for vps

**Comment by lazarus1331** [comment]
@tonysyu Since I'm merging this into TPJ, I wanted to get your approval before merging.  I'll be happy to give a Demo on Saturday so you have a change to see it in action before approving!

**Comment by tonysyu** [comment]
Thanks for tackling this!

I'm not sure I understand why this PR includes commits from Matt, but I think that's just a git housekeeping issue more than anything else. I can help with a rebase to clean those out if needed/desired.

I'm a bit confused about the use of both vagrant and docker. It's not my area of expertise, but I understood them to be different solutions for the same problem. Could you describe the roles that they play here?
